Add note on cache growth for self-hosted GitHub runners (#5757)
astral-sh/uv: docs/guides/integration/github.md

    If using `uv pip`, use `requirements.txt` instead of `uv.lock` in the cache key.

!!! warning

    [post-job-hook]: https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/running-scripts-before-or-after-a-job

>     When using non-ephemeral self-hosted runners the default cache location can grow over time,
>     becoming problematic. In such cases its possible to set the caching location relative to the

Reviewer: ```suggestion
    When using non-ephemeral self-hosted runners and the default cache directory, the cache can grow over time 
    and become problematic. In such cases, it's recommend to move the cache to inside the
```
```

    GitHub Workspace and clean it once the job finishes using a [Post Job Hook][post-job-hook].

    ```yaml
    install_job:
      env:
        # Configure a relative location for the uv cache
        UV_CACHE_DIR: ${{ github.workspace }}/.cache/uv

>       # ... restore uv cache ...

Reviewer: What is being referred to here?

Author: Good question, it was a callout to the previous section https://docs.astral.sh/uv/guides/integration/github/#caching

Reviewer: But would you actually restore the cache in this situation? My understanding is the cache is deleted at the end of each job so what is there to persist and restore? (This confusion has been why this pull request has been lingering, sorry!)

Author: Ah I see. That was meant to be optional step (if desired). We can easily remove it if its confusing.

Reviewer: ```suggestion
```

    ```

    Using a post job hook requires setting the `ACTIONS_RUNNER_HOOK_JOB_STARTED` environment
    variable on the self-hosted runner to the path of a cleanup script such as the one shown below.

>     ```sh

Reviewer: ```suggestion
    ```sh title="clean-workspace.sh"
```
